buildSrc/src/main/kotlin/com/bmuschko/gradle/docker/PublishingPlugin.kt

```diff
@@ -0,0 +1,137 @@
+                                url.set("https://github.com/cdancy")
+                            }
+                            developer {
+                                id.set("orzeh")
+                                name.set("Łukasz Warchał")
+                                url.set("https://github.com/orzeh")
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private
+    fun Project.configureBintrayExtension() {
+        configure<BintrayExtension> {
+            user = resolveProperty("BINTRAY_USER", "bintrayUser")
+            key = resolveProperty("BINTRAY_KEY", "bintrayKey")
+            setPublications("mavenJava")
+            publish = true
+
+            pkg(closureOf<BintrayExtension.PackageConfig> {
+                repo = "gradle-plugins"
+                name = "com.bmuschko:gradle-docker-plugin"
+                desc = "Gradle plugin for managing Docker images and containers."
+                websiteUrl = "https://github.com/bmuschko/${project.name}"
+                issueTrackerUrl = "https://github.com/bmuschko/${project.name}/issues"
+                vcsUrl = "https://github.com/bmuschko/${project.name}.git"
+                setLicenses("Apache-2.0")
+                setLabels("gradle", "docker", "container", "image", "lightweight", "vm", "linux")
+                publicDownloadNumbers = true
+                githubRepo = "bmuschko/${project.name}"
+
+                version(closureOf<BintrayExtension.VersionConfig> {
+                    released = SimpleDateFormat("MM/dd/yyyy").format(Date())
+                    vcsTag = "v$project.version"
+                    setAttributes(mapOf("gradle-plugin" to listOf("com.bmuschko.docker-remote-api:${project.group}:${project.name}",
+                            "com.bmuschko.docker-java-application:${project.group}:${project.name}",
+                            "com.bmuschko.docker-spring-boot-application:${project.group}:${project.name}")))
+
+                    gpg(closureOf<BintrayExtension.GpgConfig> {
+                        sign = true
+                        passphrase = resolveProperty("GPG_PASSPHRASE", "gpgPassphrase")
+                    })
+                    mavenCentralSync(closureOf<BintrayExtension.MavenCentralSyncConfig> {
+                        sync = true
+                        user = resolveProperty("MAVEN_CENTRAL_USER_TOKEN", "mavenCentralUserToken")
+                        password = resolveProperty("MAVEN_CENTRAL_PASSWORD", "mavenCentralPassword")
+                        close = "1"
+                    })
+                })
+            })
+        }
+    }
+
+    private
+    fun Project.resolveProperty(envVarKey: String, projectPropKey: String): String {
+        val propValue = System.getenv()[envVarKey]
+
+        if(propValue != null) {
+            return propValue
+        }
+
+        return findProperty(projectPropKey).toString()
+    }
```

Comment:
Shouldn't this function return `String?` instead. It could happen the property isn't set at all.

It could then be simplified to:
```kotlin
fun Project.resolveProperty(envVarKey: String, projectPropKey: String): String? =
    System.getenv()[envVarKey] ?: findProperty(projectPropKey)?.toString()
```

Reply:
Yes, absolutely. Thanks for the pointer.